projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js

```diff
@@ -0,0 +1,433 @@
+			return (
+				<div key={ index } className="videopress-block-tracks-editor__track-list-track">
+					<span>{ track.label }</span>
+					<div className="videopress-block-tracks-editor__track-list-track-delete">
+						<Button
+							isLink
+							isDestructive
+							onClick={ () => {
+								onDeleteTrack( index );
+							} }
+							aria-label={ sprintf(
+								/* translators: %s: Label of the video text track e.g: "French subtitles" */
+								__( 'Delete %s', 'jetpack' ),
+								track.label
+							) }
+							disabled={ isDeletingTrackIndex === index }
+						>
+							{ isDeletingTrackIndex === index
+								? __( 'Deleting', 'jetpack' )
+								: __( 'Delete', 'jetpack' ) }
+						</Button>
+					</div>
+				</div>
+			);
+		} );
+	}
+
+	return (
+		<MenuGroup
+			label={ __( 'Text tracks', 'jetpack' ) }
+			className="videopress-block-tracks-editor__track-list"
+		>
+			{ content }
+		</MenuGroup>
+	);
+}
+
+function SingleTrackEditor( { track, guid, onChange, onClose, onCancel, trackExists } ) {
+	const [ errorMessage, setErrorMessage ] = useState();
+	const [ isSavingTrack = false, setIsSavingTrack ] = useState();
+	const { label = '', srcLang = '', kind = DEFAULT_KIND } = track;
+
+	const fileName = track.tmpFile ? track.tmpFile.name : '';
+
+	const mediaUpload = useSelect( select => {
+		return select( blockEditorStore ).getSettings().mediaUpload;
+	}, [] );
+
+	if ( ! mediaUpload ) {
+		return null;
+	}
+
+	return (
+		<NavigableMenu>
+			<div className="videopress-block-tracks-editor__single-track-editor">
+				<span className="videopress-block-tracks-editor__single-track-editor-label">
+					{ __( 'Edit track', 'jetpack' ) }
+				</span>
+				<div className="videopress-block-tracks-editor__single-track-editor-edit-file-label">
+					<span>{ __( 'File', 'jetpack' ) }:</span>
+					{ '' !== fileName && (
+						<span className="videopress-block-tracks-editor__single-track-editor-edit-file-label-name">
+							<strong>{ fileName }</strong>
+						</span>
+					) }
+					<MediaUploadCheck>
+						<FormFileUpload
+							onChange={ event => {
+								const files = event.target.files;
+
+								if ( ! files.length > 0 ) {
+									return;
+								}
+
+								track.tmpFile = files[ 0 ];
+								onChange( track );
+							} }
+							accept=".vtt,text/vtt"
+							render={ ( { openFileDialog } ) => {
+								return (
+									<Button
+										isLink
+										onClick={ () => {
+											openFileDialog();
+										} }
+									>
+										{ '' === fileName
+											? __( 'Select track', 'jetpack' )
+											: __( 'Change track', 'jetpack' ) }
+									</Button>
+								);
+							} }
+							disabled={ isSavingTrack }
+						/>
+					</MediaUploadCheck>
+				</div>
+				<div className="videopress-block-tracks-editor__single-track-editor-label-language">
+					<TextControl
+						/* eslint-disable jsx-a11y/no-autofocus */
+						autoFocus
+						/* eslint-enable jsx-a11y/no-autofocus */
```

Comment:
Why do we need to focus here?

Reply:
🤔 I'll try it w/o it, this was copied from the core component.